include/seqan3/range/view/complement.hpp

```diff
@@ -48,20 +48,24 @@ namespace seqan3::view
 {
 
 /*!\brief A view that converts a range of nucleotides to their complement.
- * \param input_range The range you wish to convert, must satisfy seqan3::input_range_concept and the value_type
- * must satisfy seqan3::nucleotide_concept.
+ * \tparam irng_t The type of the range being processed. See below for requirements. [template parameter is omitted in pipe notation]
+ * \param irange The range being processed. [parameter is omitted in pipe notation]
+ * \returns A range of converted elements. See below for the properties of the returned range.
  * \ingroup view
- * \details
+ *
+ * Calls seqan3::nucleotide_concept::complement() on every element of the input range.
+ *
  * \par View properties
- * * view type: same as input_range
- * * value type: remove_reference_t<value_type_t<input_range>>
- * * `const` iterable: yes
- * \par Complexity
- * Linear in the size if the input range (\f$O(n)\f$).
- * \par Exceptions
- * Strong exception guarantee (does not modify data).
- * \par Thread safety
- * Does not modify data.
+ *
+ * |                     | `irng_t` (range input type)   | `rrng_t` (range return type)                                                                     |
+ * |---------------------|-------------------------------|--------------------------------------------------------------------------------------------------|
+ * | range               | seqan3::input_range_concept   | seqan3::view_concept + all range concepts met by `irng_t`                                        |
+ * | `range_reference_t` | seqan3::nucleotide_concept    | `std::remove_reference_t<range_reference_t<irng_t>>`,  *usually* == `range_value_type_t<irng_t>` |
```

Comment:
Well that depends entirely on the view. If you convert a range of dna5 to dna4, than of course the `range_value_type_t` of the input is dna5 and the `range_value_type_t` of the output is dna4.
But more important than the `range_value_type_t` is the `range_reference_t`, because (*it) and operator[] (all operations used really) return the `range_reference_t`, not the `range_value_type_t`. Now this is more interesting, because normally (say on a vector of dna5), the `range_reference_t` is `dna5 &`. If you apply `ranges::view::reverse` the reference type is preserved fully, i.e. the `range_reference_t<rrng_t>` is also `dna5 &`. This means among other things that you can use the reference value to modify the positions in the original range (unless of course the original range was `const`, in which case the reference type would have been `dna5 const &`).

On the other hand, `view::complement` of course cannot preserve regular references to the original range, because it produces new values of the nucleotide type (the complement value). That's why the `range_reference_t<rrng_t>` in this case is equal to `range_reference_t<irngt_t>` **but with the reference removed** (which may or may not be identical to the `range_value_type_t<irng_t>`). This means you cannot use the elements in the returned range to modify the original range.
*Note that you can achieve a modifying-the-original-behaviour even here, if you create a proxy type that looks like the value_type, but contains references to the original which it "re-complements" and writes back when modified. But that is unrelated to the rest and I didn't see it as useful for this implementation.*

I will add a section to the general view documentation that tries to explain this.